src/Concerns/ManagesUsageBilling.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Laravel\Cashier\Concerns;
+
+use Illuminate\Support\Collection;
+use Stripe\Billing\MeterEvent;
+
+trait ManagesUsageBilling
+{
+    /**
+     * Get all the defined billing meters.
+     *
+     * @param  array  $options
+     * @param  array  $requestOptions
+     * @return \Illuminate\Support\Collection
+     */
+    public function meters(array $options = [], array $requestOptions = []): Collection
+    {
+        return new Collection($this->stripe()->billing->meters->all($options, $requestOptions)->data);
+    }
+
+    /**
+     * Report usage for a metered product.
+     *
+     * @param  string  $meter
+     * @param  int  $quantity
+     * @param  string|null  $price
+     * @param  array  $options
+     * @param  array  $requestOptions
+     * @return \Stripe\Billing\MeterEvent
+     */
+    public function reportMeterEvent(
+        string $meter,
+        int $quantity = 1,
+        array $options = [],
+        array $requestOptions = []
+    ): MeterEvent {
+        $this->assertCustomerExists();
+
+        return $this->stripe()->billing->meterEvents->create([
+            'event_name' => $meter,
+            'payload' => [
+                'value' => $quantity,
+                'stripe_customer_id' => $this->stripeId(),
+            ],
+            ...$options,
+        ], $requestOptions);
+    }
+
+    /**
+     * Get the usage records for a meter using its ID.
+     *
+     * @param  string  $meterId
+     * @param  array  $options
+     * @param  array  $requestOptions
+     * @return \Illuminate\Support\Collection
+     */
+    public function meterEventSummaries(string $meterId, array $options = [], array $requestOptions = []): Collection
```

Comment:
The `reportMeterEvent` uses `$meter` - this uses `$meterId`? Should they also both support Backed Enums?

Reply:
So, the $meter is the event name (the meter has to be configured to listen to a specific event name), but for the summaries endpoint, we need the meter ID that's created with the meter, different from the event_name. It could be a backed enum, but I envision someone using them by saving the value for their event names and meter IDs in config files or a DB if they are created dynamically.

They could be Enums, we could also have a data class that houses both the ID and the Event Name.